cypress/integration/resources.spec.js

```diff
@@ -265,14 +265,20 @@ describe('resources', () => {
 
     cy.findAllByTestId(UPVOTE_COUNT).first().invoke('text').as('currentUpVoteCountText');
 
-    cy.findAllByTestId(UPVOTE_BUTTON).first().click({ force: true });
-    cy.wait('@upvote');
+    cy.findAllByTestId(UPVOTE_BUTTON).first().as('upvoteButton');
```

Comment:
I think you can simplify things by:

1) Making two separate tests for upvote and downvote.
2) Add a different testid if the button is active
3) Check that the different testid renders after click and that the vote count went up
4) Presuming you follow my critique in the other comment, you can ditch the test regarding the actual response and instead just reload the page after a vote to assert the vote still exists. This works out because the initial state will derive from the response and a reload ensures it's not the previous client state tick.

Reply:
Just so we are on the same page before I jump ahead:
1. That would require persisting cookie as I did before, correct?

3. So This is a little tricky provided that the test strategy is staying the same (grab the first resource, upvote first, then downvote). The issue is the initial upvoting part since there is no guarantee of the initial vote direction so there should be some conditional logic by checking whether the button is active then to assert for current upvote count +1 or -1. 

Can you write a conditional logic by checking whether certain testid renders? I am not too familiar with having conditional flow in Cypress test.

So it just dawned on me that I could utilize the response that I got from API call. Say I clicked upvote and in the response that I got,  if `current_vote_direction` is `upvote` that means the upvote count will increment by 1 whereas if it is `null`, the upvote count will decrement by 1. 

Let me know what you think.

Reply:
1) No, you can use `cy.login()` per test.

2) I don't mean to add the conditional logic in Cypress, but in the actual source code. You can totally add whatever logic you need, to any source code file, surrounding `data-testid`. Use it to your advantage to make the test code as simple as possible.

Reply:
So if I understood this correctly, we would have data-testid `UPVOTE_BUTTON` or `UPVOTE_BUTTON_ACTIVE`(something to that effect) depending on the button's active status.

Going back to line 268, let's say we leave it as is and the button was clicked. Now that the button should have `data-testid=UPVOTE_BUTTON_ACTIVE`. How can I grab the same button since there is no guarantee that ` cy.findAllByTestId(UPVOTE_BUTTON_ACTIVE).first()` is going to target the same button?